opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -842,24 +894,15 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 			"size")
 
 		// Detach the disks
-		var diskID int
 		for _, diskIf := range toDetach {
 			diskConfig := diskIf.(map[string]interface{})
 
-			imageID := diskConfig["image_id"].(int)
-			if imageID == -1 {
+			if diskConfig["image_id"].(int) == -1 &&
+				len(diskConfig["type"].(string)) == 0 {
```

Comment:
can you add a comment here explaining why you check the type to "0" ?

Reply:
Yes, I did this because this test goal is to ignore a disk.
And we want to ignore a disk with no `image_id` (-1 by default) and with no disk `type` defined because it's an empty disk section.

I'll add a comment and maybe some logging information

Reply:
better ?

Reply:
yes so it means `type = 0` is not a valid type ? Type values start at 1 ?